x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/DeprecationIndexingComponent.java

```diff
@@ -131,11 +136,27 @@ private BulkProcessor getBulkProcessor(Client client, Settings settings) {
         public void beforeBulk(long executionId, BulkRequest request) {}
 
         @Override
-        public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {}
+        public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+            long numberOfActions = request.numberOfActions();
+            if (logger.isTraceEnabled()) {
+                logger.trace(
+                    "indexed [{}] deprecation documents into [{}]",
+                    numberOfActions,
+                    Arrays.stream(response.getItems()).map(BulkItemResponse::getIndex).distinct().collect(Collectors.joining(","))
```

Comment:
nit: we could use 
```
logger.trace(
                    "indexed [{}] deprecation documents into [{}]",
                    request::numberOfActions,
()->                 Arrays.stream(response.getItems()).map(BulkItemResponse::getIndex).distinct().collect(Collectors.joining(","))
```
passing parameters suppliers instead of guarding with `logger.isTraceEnabled()`
I understand that this pattern is common though - we probably don't have a elasticsearch wide convention for this